Makefile

```diff
@@ -74,15 +74,24 @@ test:
 
 # This target is used to run Go tests that require docker runtime.
 # Some tests, like pipeline tests, need to have docker available to run.
+# It by default tests all the available runtimes. If you only want to test
+# against one specific runtime, you can set environment variable KPT_E2E_FN_RUNTIME.
+# e.g. KPT_E2E_FN_RUNTIME=docker
```

Comment:
will the tests complain if a runtime is not available on the machine ? (like on my mac, I don't have podman).

Thinking from a kpt developers perspective:
- in my dev setup, I would like to run all tests with one fn-runtime (sensible defaults for all platforms, but should be override with a personal setting in Makefile that can be preserved without checking in)
- in CI, tests should run for all the supported platforms
- Bonus would be: since these tests take the longest time, should run only if the change affects runtime

Reply:
Created https://github.com/GoogleContainerTools/kpt/pull/3194